### PR TITLE
[Feature] NVFP4 dispatch for fused RoPE quantization

### PR DESCRIPTION
**Pull request by namgyu-youn**
## Purpose

Add NVFP4 dispatch to `fused_indexer_q_rope_quant` (fused RoPE). NVFP4 is NVIDIA's proprietary FP4 format used by FlashInfer CUTLASS backends on Blackwell GPUs.

This PR adds:
- `_fused_indexer_q_rope_nvfp4_kernel`: Triton kernel (mirrors MXFP4 kernel; one thread owns each 16-element
block, no warp shuffle needed)
- `_quantize_nvfp4_block`: Triton helper (amax-based scale as float8e4nv, uses tl.div_rn for the quantization
step)
- `use_nvfp4: bool = False`: Dispatch branch in `fused_indexer_q_rope_quant`

This does not wire downstream (`sparse_attn_indexer.py` cast branch, `attention.py` flag propagation).

### Why this is needed

- Part of #45668, "NVFP4 Indexer" item on the MiniMax M3 roadmap
- FlashInfer CUTLASS NVFP4 kernels need an NVFP4-quantized Q, and no NVFP4 path exists in-tree — this PR adds it, fused from day one.
- Quant cost is +2–3 µs vs MXFP4 at decode sizes (benchmark in comments); the payoff lands downstream in
the NVFP4 logits matmul (follow-up, not this PR).

## Test Plan

```bash
# Kernel correctness — compares Triton NVFP4 output against the unfused reference
pytest -sv tests/kernels/test_fused_indexer_q_rope_quant.py

# Kernel benchmark — FP8 vs MXFP4 vs NVFP4, DeepSeek-V4 indexer shapes
python benchmarks/kernels/benchmark_fused_indexer_q_rope_quant.py
```

Both run on RTX PRO 6000 Blackwell (SM120): test 50 passed / 10 skipped (skips = CuTe-DSL NVFP4 variant, out of scope); benchmark results in [this comment](https://github.com/vllm-project/vllm/pull/46031#issuecomment-4899616125).

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by namgyu-youn** [comment]
Kernel benchmark results (command in the Test Plan). RTX PRO 6000 Blackwell (SM120, 96 GB), torch 2.11.0+cu130, Triton path.

| num_tokens | FP8 (µs) | MXFP4 (µs) | NVFP4 (µs) | NVFP4 / MXFP4 |
| --- | --- | --- | --- | --- |
| 1 | 6.85 | 8.19 | 10.24 | 1.25× |
| 4 | 7.49 | 8.19 | 10.94 | 1.34× |
| 16 | 8.16 | 9.50 | 11.58 | 1.22× |
| 64 | 9.54 | 10.94 | 13.63 | 1.25× |
| 256 | 17.66 | 21.79 | 32.74 | 1.50× |
| 1024 | 44.32 | 59.36 | 87.36 | 1.47× |
| 4096 | 153.89 | 179.49 | 292.13 | 1.63× |
| 16384 | 597.28 | 666.91 | 1114.08 | 1.67× |

The NVFP4 path is currently slower than MXFP4: ~1.25× (+2–3 µs) at decode-size M, up to 1.67× at 16K. It is not the correctly-rounded division: re-benchmarking with per-element `tl.div_rn` replaced by block-wise reciprocal + multiply (perf probe only, changes numerics) recovered just 4% at 16K. The likely cause is structural — 16-element blocks do 2× the amax reductions and scale stores of MXFP4's 32-element blocks. I plan to look at block-granularity restructuring as a follow-up; this PR's claim is the dispatch + bit-exact correctness.

<details>
<summary>Full benchmark log</summary>

```
  num_tokens=     1    fp8:     6.85 us      3.6 GB/s
  num_tokens=     1  mxfp4:     8.19 us      2.6 GB/s
  num_tokens=     1  nvfp4:    10.24 us      2.1 GB/s
  num_tokens=     4    fp8:     7.49 us     13.3 GB/s
  num_tokens=     4  mxfp4:     8.19 us     10.3 GB/s
  num_tokens=     4  nvfp4:    10.94 us      7.8 GB/s
  num_tokens=    16    fp8:     8.16 us     49.0 GB/s
  num_tokens=    16  mxfp4:     9.50 us     35.6 GB/s
  num_tokens=    16  nvfp4:    11.58 us     29.5 GB/s
  num_tokens=    64    fp8:     9.54 us    167.6 GB/s
  num_tokens=    64  mxfp4:    10.94 us    123.6 GB/s
  num_tokens=    64  nvfp4:    13.63 us    100.4 GB/s
  num_tokens=   256    fp8:    17.66 us    361.9 GB/s
  num_tokens=   256  mxfp4:    21.79 us    248.2 GB/s
  num_tokens=   256  nvfp4:    32.74 us    167.2 GB/s
  num_tokens=  1024    fp8:    44.32 us    576.9 GB/s
  num_tokens=  1024  mxfp4:    59.36 us    364.5 GB/s
  num_tokens=  1024  nvfp4:    87.36 us    250.7 GB/s
  num_tokens=  4096    fp8:   153.89 us    664.6 GB/s
  num_tokens=  4096  mxfp4:   179.49 us    482.2 GB/s
  num_tokens=  4096  nvfp4:   292.13 us    299.8 GB/s
  num_tokens= 16384    fp8:   597.28 us    684.9 GB/s
  num_tokens= 16384  mxfp4:   666.91 us    519.1 GB/s
  num_tokens= 16384  nvfp4:  1114.08 us    314.5 GB/s
fused-indexer-q-rope-quant:
   num_tokens    FP8 (us)  MXFP4 (us)   NVFP4 (us)
0         1.0    6.848000    8.192000    10.240000
1         4.0    7.488000    8.192000    10.944000
2        16.0    8.160000    9.504000    11.584000
3        64.0    9.536000   10.944000    13.632000
4       256.0   17.664000   21.792000    32.736000
5      1024.0   44.319998   59.360001    87.360002
6      4096.0  153.888002  179.488003   292.127997
7     16384.0  597.280025  666.912019  1114.079952
```

</details>

<details>
<summary>div_rn perf probe log (reciprocal-multiply variant)</summary>

```
  num_tokens=     1    fp8:     6.85 us      3.6 GB/s
  num_tokens=     1  mxfp4:     8.19 us      2.6 GB/s
  num_tokens=     1  nvfp4:    10.24 us      2.1 GB/s
  num_tokens=     4    fp8:     7.46 us     13.4 GB/s
  num_tokens=     4  mxfp4:     8.19 us     10.3 GB/s
  num_tokens=     4  nvfp4:    10.24 us      8.4 GB/s
  num_tokens=    16    fp8:     8.19 us     48.8 GB/s
  num_tokens=    16  mxfp4:     9.54 us     35.4 GB/s
  num_tokens=    16  nvfp4:    11.58 us     29.5 GB/s
  num_tokens=    64    fp8:     9.50 us    168.1 GB/s
  num_tokens=    64  mxfp4:    10.94 us    123.6 GB/s
  num_tokens=    64  nvfp4:    13.63 us    100.4 GB/s
  num_tokens=   256    fp8:    17.66 us    361.9 GB/s
  num_tokens=   256  mxfp4:    21.79 us    248.2 GB/s
  num_tokens=   256  nvfp4:    32.74 us    167.2 GB/s
  num_tokens=  1024    fp8:    44.32 us    576.9 GB/s
  num_tokens=  1024  mxfp4:    58.66 us    368.8 GB/s
  num_tokens=  1024  nvfp4:    85.31 us    256.7 GB/s
  num_tokens=  4096    fp8:   153.70 us    665.4 GB/s
  num_tokens=  4096  mxfp4:   179.49 us    482.2 GB/s
  num_tokens=  4096  nvfp4:   281.30 us    311.4 GB/s
  num_tokens= 16384    fp8:   597.28 us    684.9 GB/s
  num_tokens= 16384  mxfp4:   667.94 us    518.3 GB/s
  num_tokens= 16384  nvfp4:  1070.40 us    327.3 GB/s
fused-indexer-q-rope-quant:
   num_tokens    FP8 (us)  MXFP4 (us)   NVFP4 (us)
0         1.0    6.848000    8.192000    10.240000
1         4.0    7.456000    8.192000    10.240000
2        16.0    8.192000    9.536000    11.584000
3        64.0    9.504000   10.944000    13.632000
4       256.0   17.664000   21.792000    32.736000
5      1024.0   44.319998   58.656000    85.312001
6      4096.0  153.696001  179.488003   281.296000
7     16384.0  597.280025  667.936027  1070.400000
```

</details>

*Disclosure: benchmarking and analysis done with AI assistance (Claude); I reviewed and ran everything.*

**Comment by namgyu-youn** [comment]
cc @WoosukKwon @zyongye @mgoin @akii96 who working on fused/quantized kernel module.

Could you please look into this PR? It supports NVFP4 dispatch for fused RoPE quantization.
